### PR TITLE
Dump crash dumps into a different directory and include version in reports

### PR DESCRIPTION
**Pull request by ZehMatt**

### CODE REVIEW

File: src/openrct2/platform/Crash.cpp [hunk 1]
```diff
@@ -320,7 +322,11 @@ static std::wstring GetDumpDirectory()
 {
     char userDirectory[MAX_PATH];
     platform_get_user_directory(userDirectory, nullptr, sizeof(userDirectory));
-    auto result = String::ToWideChar(userDirectory);
+
+    auto path = Path::Combine(userDirectory, "crashes");
```
Comment (IntelOrca) [inline src/openrct2/platform/Crash.cpp:326]: Our directory names are usually singular.

Comment (ZehMatt) [reply src/openrct2/platform/Crash.cpp]: Not quite but I changed.

File: src/openrct2/platform/Crash.cpp [hunk 2]
```diff
@@ -320,7 +322,11 @@ static std::wstring GetDumpDirectory()
 {
     char userDirectory[MAX_PATH];
     platform_get_user_directory(userDirectory, nullptr, sizeof(userDirectory));
-    auto result = String::ToWideChar(userDirectory);
+
+    auto path = Path::Combine(userDirectory, "crash");
```
Comment (Gymnasiast) [inline src/openrct2/platform/Crash.cpp:326]: We have enums for these types of directory.

Called like so: env->GetDirectoryPath(OpenRCT2::DIRBASE::OPENRCT2, OpenRCT2::DIRID::CRASH);

**Review by Gymnasiast — APPROVED** [review]
All my points have been addressed. :+1: 
Leaving it to @IntelOrca now.

**Review by Broxzier — APPROVED** [review]

File: src/openrct2/platform/Crash.cpp [hunk 3]
```diff
@@ -318,9 +320,10 @@ static bool OnCrash(
 
 static std::wstring GetDumpDirectory()
 {
-    char userDirectory[MAX_PATH];
-    platform_get_user_directory(userDirectory, nullptr, sizeof(userDirectory));
-    auto result = String::ToWideChar(userDirectory);
+    auto env = GetContext()->GetPlatformEnvironment();
```
Comment (Gymnasiast) [inline src/openrct2/platform/Crash.cpp:323]: I think this should read

```suggestion
    auto env = OpenRCT2::GetContext()->GetPlatformEnvironment();
```

**Review by IntelOrca — APPROVED** [review]

File: src/openrct2/platform/Crash.cpp (hunk 3 above)
Comment (ZehMatt) [reply src/openrct2/platform/Crash.cpp]: Oddly everything except MSVC built this, on my end it also did but yeah.

**Pull request merged**